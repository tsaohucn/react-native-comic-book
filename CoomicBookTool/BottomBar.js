import React from 'react'
import { 
  View, 
  StyleSheet, 
  Text, 
  Animated, 
  Dimensions, 
  TouchableWithoutFeedback 
} from 'react-native'
import { Icon } from 'react-native-elements'
import ComicBookBrightness from './ComicBookBrightness'

const BottomBar = ({animatedBottomBarY, onClickChapterBar, onClickProgressBar, onClickConfigBar }) => (
  <Animated.View style={[styles.bottomToolBarAnimated,{
    transform: [
      { translateY: animatedBottomBarY }
      ]}]}
    >
      <TouchableWithoutFeedback>
        <View style={styles.bottomToolBarIconView}>
          <Icon
            name='radio-tower'
            type='material-community'
            color='white'
          />
          <Text style={styles.iconText}>發彈幕</Text> 
        </View> 
      </TouchableWithoutFeedback>
      <TouchableWithoutFeedback
        onPress={onClickChapterBar} 
      >
        <View style={styles.bottomToolBarIconView}>
          <Icon
            name='view-headline'
            type='material-community'
            color='white'
          />
          <Text style={styles.iconText}>目錄</Text> 
        </View>
      </TouchableWithoutFeedback>
      <TouchableWithoutFeedback
        onPress={onClickProgressBar} 
      >
        <View style={styles.bottomToolBarIconView}>
          <Icon
            name='toggle-switch'
            type='material-community'
            color='white'
          />
          <Text style={styles.iconText}>進度</Text> 
        </View>
      </TouchableWithoutFeedback> 
      <ComicBookBrightness/> 
      <TouchableWithoutFeedback
        onPress={onClickConfigBar} 
      >
        <View style={styles.bottomToolBarIconView}>
          <Icon
            name='video-input-component'
            type='material-community'
            color='white'
          />
        <Text style={styles.iconText}>設定</Text> 
      </View>
    </TouchableWithoutFeedback> 
  </Animated.View>
)

const { width, height } = Dimensions.get('window')

const styles = StyleSheet.create({
  iconText: {
    color: 'white'
  },
  bottomToolBarAnimated: {
    position: 'absolute',
    bottom: 0,
    width,
    height: 50,
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
    backgroundColor: 'rgba(52, 52, 52, 0.8)',
  },
  bottomToolBarIconView: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  }
})

export default BottomBar