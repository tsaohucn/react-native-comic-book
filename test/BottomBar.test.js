// __tests__/BottomBar.js
import 'react-native'
import React from 'react'
import BottomBar from '../CoomicBookTool/BottomBar'

import renderer from 'react-test-renderer'

jest.mock('react-native-elements', () => ({
	Icon: 'Icon'
}))

describe('Testing BottomBar component', () => {

	it('renders correctly with no props', () => {
		jest.resetModules()
	  jest.doMock('Platform', () => ({
	    OS: 'android',
	  }))

	  const a = require('Platform').default
		const props = {

		}
	  const tree = renderer.create(
	    <BottomBar {...props}/>
	    ).toJSON();
	  expect(tree).toMatchSnapshot();
	})

	it('renders correctly with animatedBottomBarY value', () => {
		const props = {
			animatedBottomBarY: 100
		}
	  const tree = renderer.create(
	    <BottomBar {...props}/>
	    ).toJSON();
	  expect(tree).toMatchSnapshot();
	})

})