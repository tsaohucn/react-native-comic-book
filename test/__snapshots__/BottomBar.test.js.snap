// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing BottomBar component renders correctly with animatedBottomBarY value 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(52, 52, 52, 0.8)",
      "bottom": 0,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "space-around",
      "position": "absolute",
      "transform": Array [
        Object {
          "translateY": 100,
        },
      ],
      "width": 750,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID={undefined}
  >
    <Icon
      color="white"
      name="radio-tower"
      type="material-community"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    >
      發彈幕
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID={undefined}
  >
    <Icon
      color="white"
      name="view-headline"
      type="material-community"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    >
      目錄
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID={undefined}
  >
    <Icon
      color="white"
      name="toggle-switch"
      type="material-community"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    >
      進度
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID={undefined}
  >
    <Icon
      color="white"
      name="shield-half-full"
      type="material-community"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    >
      夜間
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID={undefined}
  >
    <Icon
      color="white"
      name="video-input-component"
      type="material-community"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    >
      設定
    </Text>
  </View>
</View>
`;

exports[`Testing BottomBar component renders correctly with no props 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(52, 52, 52, 0.8)",
      "bottom": 0,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "space-around",
      "position": "absolute",
      "transform": Array [
        Object {
          "translateY": 1334,
        },
      ],
      "width": 750,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID={undefined}
  >
    <Icon
      color="white"
      name="radio-tower"
      type="material-community"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    >
      發彈幕
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID={undefined}
  >
    <Icon
      color="white"
      name="view-headline"
      type="material-community"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    >
      目錄
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID={undefined}
  >
    <Icon
      color="white"
      name="toggle-switch"
      type="material-community"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    >
      進度
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID={undefined}
  >
    <Icon
      color="white"
      name="shield-half-full"
      type="material-community"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    >
      夜間
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID={undefined}
  >
    <Icon
      color="white"
      name="video-input-component"
      type="material-community"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    >
      設定
    </Text>
  </View>
</View>
`;
